src/GitHub.Exports/Services/VSServices.cs

```diff
@@ -68,6 +72,66 @@ public void ActivityLogWarning(string message)
             }
         }
 
+        /// <summary>
+        /// There doesn't appear to be a command that directly opens a target repo.
+        /// Our workaround is to create, open and delete a solution in the repo directory.
+        /// This triggers an event that causes the target repo to open. ;)
+        /// </summary>
+        /// <param name="directory">The target repo directory. </param>
+        public bool TryOpenRepository(string directory)
+        {
+            try
+            {
+                // Use a prefix (~$) that is defined in the default VS gitignore.
+                var name = "~$GitHubVSTemp$~";
+
+                // If .vs\ exists, create the dummy sln there.
+                var vsDir = Path.Combine(directory, ".vs");
+                if(Directory.Exists(vsDir))
+                {
+                    directory = vsDir;
+                }
+
+                var file = Path.Combine(directory, name + ".sln");
+
+                // If something went wrong last time, the solution file might already exist.
+                if(File.Exists(file))
+                {
+                    File.Delete(file);
+                }
+
+                dte.Solution.Create(directory, name);
+                dte.Solution.Close(false);
+
+                // Clean up the dummy solution file.
+                // Try a couple of times incase antivirus app has take a lock.
+                TryFileDelete(file, 2, 500);
+
+                return true;
+            }
+            catch (Exception e)
+            {
+                VsOutputLogger.WriteLine("Error opening repository. {0}", e);
+                return false;
+            }
+        }
+
+        void TryFileDelete(string file, int retries, int msTimeout)
```

Comment:
This will need to be replaced by a robust and quick scrub directory method (for the `.vs\~$GitHubVSTemp$~` dir). Does such a thing exist already in the code base?